addons/im_livechat/controllers/main.py

```diff
@@ -151,7 +147,7 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
             with replace_exceptions(UserError, by=NotFound()):
                 # sudo: mail.guest - creating a guest and their member in a dedicated channel created from livechat
                 __, guest = channel.sudo()._find_or_create_persona_for_channel(
-                    guest_name=self._get_guest_name(),
+                    guest_name=channel.livechat_channel_id._get_guest_name(),
```

Comment:
Why is this moved from controller to model?
guest name is not a concept on livechat channel model.

Reply:
`anonymous_name` was used to calculate the name of the channel (`visitor` or `visitor #XXX`), which is similar to the guest names we use. I've moved the method to the channel to make it easier to use in both places.